src/Datasource/EntityTrait.php

```diff
@@ -292,9 +295,28 @@ public function &get(string $field): mixed
             return $result;
         }
 
+        if (!$fieldIsPresent && $this->requireFieldPresence) {
+            throw new MissingPropertyException([
+                'property' => $field,
+                'entity' => $this::class,
+            ]);
+        }
+
         return $value;
     }
 
+    /**
+     * Enable/disable field presence check when accessing a property.
+     *
+     * If enabled an exception will be thrown when trying to access a non-existent property.
+     *
+     * @param bool $value `true` to enable, `false` to disable.
+     */
+    public function requireFieldPresence(bool $value = true): void
```

Comment:
If this is an opt-in flow, is the plan to make this the default in the future?

Reply:
Yes I think having it enabled by default innfutir would help in writing error free code. For e.g. catch typos in field name or use of fields which haven't been fetched but are actually required.